### ISSUE TITLE
Terminal profile preferences ignored

### ISSUE THREAD

**Issue by RivenSkaye** [lapce/lapce#2812]
## Lapce Version
Nightly, built ~~yesterday~~ 37 minutes ago.
Update: just rebuilt the same commit with the stable toolchain and no config/env overrides, the problem persists

## System information
Windows 10 Pro 22H2 

## Describe the bug
Altering the settings file and creating a custom profile, then subsequently making that the default profile, is not being respected. In the case of Windows, this means that it will _always_ open powershell. This effectively bars the use of any shell other than the default and there are no menu options or commands to select a profile other than the default for a one-off launch either.

## Additional information
Config for reference. Removing the args, changing the command, pointing it to `.cmd` scripts or even just supplying `cmd.exe` as command makes no difference.

```toml
[terminal]
    font-family = "Hasklug NF"
    font-size   = 14

[terminal.profiles.msys]
    args    = ["--login", "-i"]
    command = "C:\\msys64\\usr\\bin\\bash.exe"
    env     = {MSYSTEM = "MINGW64", CHERE_INVOKING = "1", MSYS2_PATH_TYPE = "inherit"}

[terminal.default-profile]
    windows = "msys"
```
![image](https://github.com/lapce/lapce/assets/14993038/e4de6223-963d-4cbe-a1bb-e6c0289d35af)

**Comment by eugenesvk** [comment]
The current nightly (nightly-8312021) doesn't seem to fix this, on Windows I still get the default powershell with this config (the font-size increase works)

```toml
[terminal]
font-size  	= 104

[terminal.default-profile]
windows = "cmd"

[terminal.profiles]
cmd    	= {"command" = "C:/Windows/System32/cmd.exe"}
```

**Comment by panekj** [comment]
works fine for me
```toml
[terminal.default-profile]
windows = "cmd"

[terminal.profiles]
cmd = { command = "C:\\Windows\\System32\\cmd.exe" }
```

**Comment by eugenesvk** [comment]
By the way, is it possible to completely disable loading a terminal not to wait for the slow powershell?

**Comment by willjsaint** [comment]
> works fine for me
> 
> ```toml
> [terminal.default-profile]
> windows = "cmd"
> 
> [terminal.profiles]
> cmd = { command = "C:\\Windows\\System32\\cmd.exe" }
> ```



> The current nightly (nightly-8312021) doesn't seem to fix this, on Windows I still get the default powershell with this config (the font-size increase works)
> 
> ```toml
> [terminal]
> font-size  	= 104
> 
> [terminal.default-profile]
> windows = "cmd"
> 
> [terminal.profiles]
> cmd    	= {"command" = "C:/Windows/System32/cmd.exe"}
> ```

neither of these work for me on 0.3.1

**Comment by willjsaint** [comment]
> > neither of these work for me on 0.3.1
> 
> that's because it's not fixed in 0.3.1

hm yea that explains it lmfao
